server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -197,7 +199,14 @@ public void writeTo(StreamOutput out) throws IOException {
         if (hasFallback) {
             new WriteableSetting(setting.fallbackSetting, type).writeTo(out);
         }
-        // We are using known types so don't need the parser
+        // Write a boolean specifying whether the parser is an instanceof writeable
+        boolean isParserWriteable = parser instanceof Writeable;
+        if (isParserWriteable = true) {
```

Comment:
1. `=` is assignment, you would need `==` here.
2. But `isParserWriteable == true` is the same as a plain boolean `isParserWriteable`